tools/cluster.py

```diff
@@ -506,15 +506,23 @@ def job_attach(cluster_name, key_file=default_key_file, job_name=None, job_tag=N
 class NotHealthyCluster(Exception): pass
 
 @named('health-check')
-def health_check(cluster_name, key_file=default_key_file, master=None, remote_user=default_remote_user, region=default_region):
-    master = master or get_master(cluster_name, region=region)
-    all_args = load_cluster_args(master, key_file, remote_user)
-    nslaves = int(all_args['slaves'])
-    minimum_percentage_healthy_slaves = all_args['minimum_percentage_healthy_slaves']
-    masters, slaves = get_active_nodes(cluster_name, region=region)
-    if nslaves == 0 or float(len(slaves)) / nslaves < minimum_percentage_healthy_slaves:
-        raise NotHealthyCluster('Not enough healthy slaves: {0}/{1}'.format(len(slaves), nslaves))
-
+def health_check(cluster_name, key_file=default_key_file, master=None, remote_user=default_remote_user, region=default_region, retries=3):
+    for i in range(retries):
```

Comment:
This retry is okay, but I think we still should make the main retry handle this kind of exception, something like this:
```python
import ssl
(...)
        except (subprocess.CalledProcessError, ssl.SSLError) as e:
            failures += 1
            log.exception('Got exception')
            last_failure = 'Exception: {}'.format(e)
```